### ISSUE TITLE
What does this nonRemovablePods mean?

### ISSUE THREAD

**Issue by jeffersonbabuyo** [kubernetes-sigs/descheduler#1528]
I am trying to use the strategy called lowNodeUtilization and it is now deployed in my dev aks cluster. I am running the descheduler via cronjob.

But when I listen to the descheduler job log events, I notice that descheduler was unable to perform an eviction due to **nonRemovablePods** and I've been searching in the documentation about what does this mean.

Has anyone encountered this issue before? How did you resolved it in your end?

```
nodeutilization.go:210] "Node is appropriately utilized" node="aks-agentpool4-41349920-vmss00000c" usage={"cpu":"3370m","memory":"6232Mi","pods":"27"} usagePercentage={"cpu":87.31,"memory":57.88,"pods":90}
nodeutilization.go:210] "Node is appropriately utilized" node="aks-agentpool4-41349920-vmss00000i" usage={"cpu":"3470m","memory":"6160Mi","pods":"22"} usagePercentage={"cpu":89.9,"memory":57.21,"pods":73.33}
nodeutilization.go:204] "Node is underutilized" node="aks-agentpool4-41349920-vmss00000m" usage={"cpu":"1774m","memory":"2791Mi","pods":"14"} usagePercentage={"cpu":45.96,"memory":25.92,"pods":46.67}
lownodeutilization.go:134] "Criteria for a node under utilization" CPU=100 Mem=50 Pods=100
lownodeutilization.go:135] "Number of underutilized nodes" totalNumber=2
lownodeutilization.go:148] "Criteria for a node above target utilization" CPU=100 Mem=70 Pods=100

lownodeutilization.go:149] "Number of overutilized nodes" totalNumber=2
nodeutilization.go:260] "Total capacity to be moved" CPU=4072 Mem=9819768422 Pods=31
nodeutilization.go:263] "Evicting pods from node" node="aks-agentpool4-41349920-vmss00000a" usage={"cpu":"3470m","memory":"7930Mi","pods":"24"}
nodeutilization.go:266] "Pods on node" node="aks-agentpool4-41349920-vmss00000a" allPods=24 nonRemovablePods=24 removablePods=0

nodeutilization.go:269] "No removable pods on node, try next node" node="aks-agentpool4-41349920-vmss00000a"
nodeutilization.go:263] "Evicting pods from node" node="aks-agentpool4-41349920-vmss000009" usage={"cpu":"3530m","memory":"7676Mi","pods":"23"}
```

**Comment by damemi** [comment]
Pods may not be removable by default for the reasons outlined here: https://github.com/kubernetes-sigs/descheduler#pod-evictions

You can set the logging level at `v=4` or greater to see the exact reasons why pods were not evicted

**Comment by jeffersonbabuyo** [comment]
@damemi - Thank you for your prompt response. I will try increasing the logLevel to 4 and listen for the descheduler once again. By the way I am installing descheduler using Helm. I will update this thread once I have found something.

**Comment by jeffersonbabuyo** [comment]
hello @damemi - So I have set the logging level to `v=4` and I can now see the reason why descheduler was unable to evict certain pod.

```
checks="[pod is a DaemonSet pod, pod has local storage and descheduler is not configured with evictLocalStoragePods]"
checks="pod is a DaemonSet pod"
checks="[pod is a DaemonSet pod, pod has system critical priority, pod has higher priority than specified priority class threshold]
```
I did make some adjustments on my descheduler configuration by allowing descheduler to evict pods with localstorage and pods that is a daemonset but still descheduler was unable to evict those pods.

```
LowNodeUtilization:
      enabled: true
      params:
        nodeResourceUtilizationThresholds:
          thresholds:
            memory: 50
          targetThresholds:
            memory: 70
          evictableNamespaces:
            exclude:
            - "kube-system"
        evictLocalStoragePods: true
        evictDaemonSetPods: true
```

**Comment by jeffersonbabuyo** [comment]
Closing this issue now as I have managed to resolved the issue by upgrading to v0.31
